CRM/Contribute/Form/ContributionBase.php

```diff
@@ -328,13 +328,11 @@ public function preProcess() {
         }
       }
 
-      if ($isMonetary &&
-        (!$isPayLater || !empty($this->_values['payment_processor']))
-      ) {
-        $this->_paymentProcessorIDs = explode(
+      if ($isMonetary) {
+        $this->_paymentProcessorIDs = array_filter(explode(
           CRM_Core_DAO::VALUE_SEPARATOR,
           CRM_Utils_Array::value('payment_processor', $this->_values)
-        );
+        ));
```

Comment:
`explode(' ', NULL);` returns `array(0 => NULL)` which loads processor key `0`(pay later) on the contribution page.

Reply:
Agree